Enterprise-CMCS/macpro-mdct-mcr
# [MDCT-2215] Lock + Release MLR reports

## Summary

This PR adds some functionality to release and lock MLR reports.

When reports are submitted they are locked, meaning they are **viewable** but not **editable**. When reports are "released", they are unlocked, and changes to the reports field data are saved in a new location in S3. (We also track these locations `reportMetadata.previousRevisions`)

This PR also cherry-picks the "submit" handler from the statusing feature. This changes the way in which we submit reports—instead of calling the update handler, we call the submit handler, which changes the requisite details in the report metadata.

### Description
Lastly, I've started adding more complete interfaces with simple validators for Dynamo data. Specifically, a new `isMLRReport` function which will cast any object to an `MLRReportMetadata` type if the correct data is in the report.


### Related ticket
[](https://qmacbis.atlassian.net/browse/MDCT-2215)

### How to test 
1. Create and submit an MLR report
2. Verify that report's metadata is set to "locked"
3. From the admin dashboard, select "release" for that report.
4. Verify that the report's metadata has a new field id, and the "previousRevisions" array has a value.
5. Enter the released report
6. Verify that option 1 has been selected for question G, and that question H is visible
7. Verify that all options for question H are selectable.

### Important updates
No changes to environment files, but this is not fully tested yet.

### Author checklist
